Corona-Warn-App/src/main/java/de/rki/coronawarnapp/greencertificate/ui/certificates/details/CovidCertificateDetailsFragment.kt

```diff
@@ -0,0 +1,172 @@
+import androidx.navigation.fragment.findNavController
+import androidx.navigation.fragment.navArgs
+import com.google.android.material.appbar.AppBarLayout
+import de.rki.coronawarnapp.R
+import de.rki.coronawarnapp.bugreporting.ui.toErrorDialogBuilder
+import de.rki.coronawarnapp.covidcertificate.test.TestCertificate
+import de.rki.coronawarnapp.databinding.FragmentCovidCertificateDetailsBinding
+import de.rki.coronawarnapp.ui.qrcode.fullscreen.QrCodeFullScreenFragmentArgs
+import de.rki.coronawarnapp.ui.view.onOffsetChange
+import de.rki.coronawarnapp.util.DialogHelper
+import de.rki.coronawarnapp.util.TimeAndDateExtensions.toDayFormat
+import de.rki.coronawarnapp.util.TimeAndDateExtensions.toShortDayFormat
+import de.rki.coronawarnapp.util.TimeAndDateExtensions.toShotTimeFormat
+import de.rki.coronawarnapp.util.di.AutoInject
+import de.rki.coronawarnapp.util.setUrl
+import de.rki.coronawarnapp.util.ui.popBackStack
+import de.rki.coronawarnapp.util.ui.viewBinding
+import de.rki.coronawarnapp.util.viewmodel.CWAViewModelFactoryProvider
+import de.rki.coronawarnapp.util.viewmodel.cwaViewModelsAssisted
+import javax.inject.Inject
+
+class CovidCertificateDetailsFragment : Fragment(R.layout.fragment_covid_certificate_details), AutoInject {
+
+    @Inject lateinit var viewModelFactory: CWAViewModelFactoryProvider.Factory
+    private val binding by viewBinding<FragmentCovidCertificateDetailsBinding>()
+    private val args by navArgs<CovidCertificateDetailsFragmentArgs>()
+    private val viewModel: CovidCertificateDetailsViewModel by cwaViewModelsAssisted(
+        factoryProducer = { viewModelFactory },
+        constructorCall = { factory, _ ->
+            factory as CovidCertificateDetailsViewModel.Factory
+            factory.create(
+                testCertificateIdentifier = args.testCertificateIdentifier
+            )
+        }
+    )
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) = with(binding) {
+        qrCodeCard.title.setText(R.string.detail_green_certificate_card_title)
+        appBarLayout.onOffsetChange { titleAlpha, subtitleAlpha ->
+            title.alpha = titleAlpha
+            subtitle.alpha = subtitleAlpha
+        }
+
+        bindTravelNoticeViews()
+        bindToolbar()
+        setToolbarOverlay()
+
+        viewModel.qrCode.observe(viewLifecycleOwner) { onQrCodeReady(it) }
+        viewModel.errors.observe(viewLifecycleOwner) { onError(it) }
+        viewModel.events.observe(viewLifecycleOwner) { onNavEvent(it) }
+        viewModel.covidCertificate.observe(viewLifecycleOwner) { it?.let { onCertificateReady(it) } }
+    }
+
+    private fun FragmentCovidCertificateDetailsBinding.onCertificateReady(
+        testCertificate: TestCertificate
+    ) {
+        qrCodeCard.subtitle.text = getString(
+            R.string.detail_green_certificate_card_subtitle,
+            testCertificate.testResultAt.toShortDayFormat(),
+            testCertificate.testResultAt.toShotTimeFormat(),
+        )
```

Comment:
@d4rken  which date should be used here ?
<img width="595" alt="Screenshot 2021-06-04 at 15 42 12" src="https://user-images.githubusercontent.com/25054729/120810665-889ad680-c54b-11eb-8f0b-2b0f1b193b39.png">

Reply:
`dr`, `testResultAt` would be correct AFAIK, I saw in another PR though that the `dr` attribute is optional now, so we need to clarify whether this is still desired and if there needs to be a fallback value :thinking: 

/cc @mlenkeit

Reply:
`sc` should be used here.

Reply:
> `sc` should be used here.

Thanks, so `sampleCollectedAt` @axelherbstreith